lib/string_tools/core_ext/string.rb

```diff
@@ -39,6 +39,44 @@ def naturalized
     scan(/[^\d\.]+|[\d\.]+/).map{|f| f.match(/^\d+(\.\d+)?$/) ? f.to_f : f }
   end
 
+  def self.natcmp(str1, str2)
```

Comment:
Cyclomatic complexity for natcmp is too high. [12/6]<br>Assignment Branch Condition size for natcmp is too high. [35.3/35]<br>Perceived complexity for natcmp is too high. [9/7]